src/listener.ts

```diff
@@ -0,0 +1,15 @@
+import { Awaitable, ClientEvents } from "discord.js";
+
+export type ListenerType = "on" | "once";
+
+/** @private */
+type MappedListeners = {
+	[Event in keyof ClientEvents]: {
+		/** @default "on" */
+		type?: ListenerType;
+		event: Event;
+		handler: (...args: ClientEvents[Event]) => Awaitable<void>;
+	};
+};
+
+export type Listener = MappedListeners[keyof MappedListeners];
```

Comment:
This would probably be better off in a `/types` directory considering this has no runtime functionality. Also, in general, try to remember to add type to the imports, e.g. `import type { Awaitable } from "discord.js"

Reply:
I'll enable [verbatimModuleSyntax](https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax) in the tsconfig